tree/tree.h

```diff
@@ -0,0 +1,47 @@
+#ifndef _TREE_H__ 
+#define _TREE_H__
```

Comment:
Hello Rafael, Just now I understood your request for the review. Now that I'm understanding more of how the github works let's  go to your code.

According to the style guide, the #define guard should contain the entire path of the project. Thus it should be:
`#ifndef  MACHINELEARNING_TREE_TREE_H_`

or as we mentioned earlier, a more c++ representation with:
`#ifndef  MACHINELEARNING_TREE_TREE_HPP_`